spring-kafka-test/src/main/java/org/springframework/kafka/test/ZooKeeperServer.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.test;
+
+import java.net.ServerSocket;
+import java.util.Properties;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.kafka.test.TestUtils;
+import org.apache.zookeeper.server.ServerConfig;
+import org.apache.zookeeper.server.ZooKeeperServerMain;
+import org.apache.zookeeper.server.quorum.QuorumPeerConfig;
+
+/**
+ * A zookeeper server to use with unit tests.
+ *
+ * @author Dmitrii Apanasevich
+ */
+public final class ZooKeeperServer {
```

Comment:
We can't accept such a breaking change. On the other hand it looks like in the end you are pursuing a `cnxnFactory.shutdown()`, which, looks like is accessed via:
```
this.zookeeper.zookeeper().shutdown();
this.zookeeper.zookeeper().getServerCnxnFactory().shutdown();
```
in the `EmbeddedKafkaBroker`.

I agree about other polishing around magic numbers and missed logs for swallowed exception, but I don't see reason in this class overhead which may lag with updates in the future Apache Kafka versions.

So, WDYT about my suggestion?

Reply:
I have no other options, so I've created [an issue](https://issues.apache.org/jira/browse/ZOOKEEPER-3248) in Apache tracker.